src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -353,11 +353,13 @@ jint ShenandoahHeap::initialize() {
   }
 
   _regions = NEW_C_HEAP_ARRAY(ShenandoahHeapRegion*, _num_regions, mtGC);
+  _affiliations = NEW_C_HEAP_ARRAY(uint8_t, _num_regions, mtGC);
```

Comment:
Should we only initialize this for generational mode? or are there calls to check affiliation in the other modes too?

Reply:
Good point.  I think it's only needed for generational mode.

Reply:
After further investigation, I see that we need _affiliation for non-generational mode to keep track of FREE status.  I've added a comment to this effect.